src/cli.rs

```diff
@@ -759,6 +777,7 @@ impl Opts {
             .filter(|(_, (name, pin))| {
                 (opts.names.is_empty() || valid_names.contains(name))
                     && (opts.update_frozen || !pin.is_frozen())
+                    && pin.can_auto_update()
```

Comment:
It turns out this logic is wrong, because it does not account for the update strategy. While such a pin cannot *update* in the sense of `Updatable::update`, it can still have a hash change in the sense of `Updatable::fetch`, which is handled in the same code path.

Moreover, given that we cannot guarantee information on a tarball pin which we do not have (e.g. before first fetch) or if the server implementation changes its mind, my suggestion would be to remove the `can_auto_update` logic again and simply let the pin do its thing (meaning that regular tarball pins will simply happen to be up to date all of the time).

Reply:
true, i didn't realize that. i've just reverted the whole commit^^